hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java

```diff
@@ -749,57 +749,86 @@ public boolean isBound(Parameter<?> parameter) {
 	@Override
 	public <T> T getParameterValue(Parameter<T> parameter) {
 		LOGGER.tracef( "#getParameterValue(%s)", parameter );
-
 		getProducer().checkOpen( false );
 
-		if ( !getParameterMetadata().containsReference( (QueryParameter) parameter ) ) {
-			throw new IllegalArgumentException( "Parameter reference [" + parameter + "] did not come from this query" );
-		}
+		final String notBoundErrorMessage = "Parameter value not yet bound : " + parameter.toString();
+		final String couldNotResolveErrorMessage = "Parameter reference [" + parameter + "] did not come from this query";
+		final String boundCheckingLogMessage = "Checking whether parameter reference " + parameter + " is bound : %s";
 
-		final QueryParameterBinding<T> binding = getQueryParameterBindings().getBinding( (QueryParameter<T>) parameter );
-		LOGGER.debugf( "Checking whether parameter reference [%s] is bound : %s", parameter, binding.isBound() );
-		if ( !binding.isBound() ) {
-			throw new IllegalStateException( "Parameter value not yet bound : " + parameter.toString() );
-		}
-		return binding.getBindValue();
+		return (T) getParameterValue(
+				(QueryParameter) parameter,
+				notBoundErrorMessage,
+				couldNotResolveErrorMessage,
```

Comment:
Do we have inconsistent whitespace usage issue above? The content might cease to align vertically correctly if we change the tab size.

Reply:
The issue is exemplified as in the following screenshot (in https://github.com/hibernate/hibernate-orm/blob/wip/6.0/hibernate-core/src/main/antlr/org/hibernate/grammars/hql/HqlParser.g4):
![Screen Shot 2020-10-29 at 10 51 27 AM](https://user-images.githubusercontent.com/8211458/97590284-d60efe00-19d4-11eb-887d-67996957c844.png)

Reply:
not sure, in IJ it looks fine